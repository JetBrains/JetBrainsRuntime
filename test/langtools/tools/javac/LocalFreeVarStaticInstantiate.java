/*
 * @test /nodynamiccopyright/
 * @bug 8322882
 * @summary Disallow attempts to access a free variable proxy field from a static method
 * @compile/fail/ref=LocalFreeVarStaticInstantiate.out -XDrawDiagnostics LocalFreeVarStaticInstantiate.java
 */

class LocalFreeVarStaticInstantiate {

    // local class in method
    static void foo(Object there) {
        class Local {
            {
                there.hashCode();
            }

            static {
                new Local();    // can't get there from here
            }

            static Runnable r = () -> {
                new Local();    // can't get there from here
            };
        }
    }

    // local class in lambda
    static Runnable foo = () -> {
        Object there = "";
        class Local {
            {
                there.hashCode();
            }

            static {
                new Local();    // can't get there from here
            }

            static Runnable r = () -> {
                new Local();    // can't get there from here
            };
        }
    };
}
