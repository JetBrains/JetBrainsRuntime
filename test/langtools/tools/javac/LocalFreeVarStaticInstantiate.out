LocalFreeVarStaticInstantiate.java:18:17: compiler.err.local.cant.be.inst.static: kindname.class, Local
LocalFreeVarStaticInstantiate.java:22:17: compiler.err.local.cant.be.inst.static: kindname.class, Local
LocalFreeVarStaticInstantiate.java:36:17: compiler.err.local.cant.be.inst.static: kindname.class, Local
LocalFreeVarStaticInstantiate.java:40:17: compiler.err.local.cant.be.inst.static: kindname.class, Local
4 errors
